### PR TITLE
Added search functionality

### PR DESCRIPTION
**Pull request by dkusznir**
@mhanlon: 
Known issues:
https://github.com/dkusznir/blocnotes/issues/26
https://github.com/dkusznir/blocnotes/issues/25

### CODE REVIEW

File: blocnotes/blocnotes/MasterViewController.m [hunk 1]
```diff
@@ -62,14 +62,45 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
...
+            else
+            {
+                object = [[self fetchedResultsController] objectAtIndexPath:indexPath];
+            }
+            
+            DetailViewController *controller = (DetailViewController *)[[segue destinationViewController] topViewController];
+            [controller setDetailItem:object];
+            controller.navigationItem.leftBarButtonItem = self.splitViewController.displayModeButtonItem;
+            controller.navigationItem.leftItemsSupplementBackButton = YES;
+            [self.searchController dismissViewControllerAnimated:YES completion:nil];
```
Comment (mhanlon) [inline blocnotes/blocnotes/MasterViewController.m]: I would strongly consider getting rid of these next three lines:
```
            [self.searchController dismissViewControllerAnimated:YES completion:nil];
            self.searchController.searchBar.text = @"";
            [self.tableView reloadData];
```
The person may need to muck around in the search result a bunch to find the one they really want, so I would, as a user, expect to be returned to my search results when I back out of my note.

**Comment by dkusznir** [comment]
@mhanlon: I completely agree. I only put that code there because I was running into a problem where if the user navigated back, the cell titles would be correct, but if tapped, they would should the wrong note content - the content would be from the most recent item in the DB. I'm fairly confident that I'm not checking (or incorrectly checking) for self.searchController.active and the wrong call is being made:
`<[self.fetchedResultsController objectAtIndexPath:indexPath]>` instead of `<[self.filteredList objectAtIndex:indexPath.row]>`

**Comment by mhanlon** [comment]
So with this fix in (removing those three lines), I get the search bar over my detail view:
blocnotesscreenshot

This may not 100% be what you want. I know the (now deprecated)`UISearchDisplayController` is a little... cleaner, in how it inserts itself into the navigation stack. But this approach isn't *too* bad... when I hit the (x) the search text clears and I'm left on the note I found... then when I hit back I go back to the list, unfiltered... if you want the old behavior, you can set the following property on your `UISearchController`: 
```
    self.searchController.hidesNavigationBarDuringPresentation = NO;
```

In both cases (turning off `.hidesNavigationBarDuringPresentation` and leaving it on by default), when I get back to the tableview of notes I don't see the problem you described, where tapping a row brings you to the wrong note... One thing you may want to check is the matching call to 
```
    self.searchController.active = NO;
```

Also, if we're being nitpicky (:)), I would make sure all the calls are consistent... if you're going to use `isActive` as the getter, use that, otherwise, the `active` reference is fine.

File: blocnotes/blocnotes/MasterViewController.m [hunk 2]
```diff
@@ -62,14 +62,45 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
...
+            controller.navigationItem.leftBarButtonItem = self.splitViewController.displayModeButtonItem;
+            controller.navigationItem.leftItemsSupplementBackButton = YES;
+        }
+        
+        else
+        {
+            NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+            NSManagedObject *object = nil;
+            
+            if (self.searchController.isActive)
```
Comment (mhanlon) [inline blocnotes/blocnotes/MasterViewController.m]: This is the nitpicky mention. I would stick with `isActive` or `active` all over the place, for your search controller. Consistency is the hobgoblin of little minds, but it sure makes reading code easier. :)

**Pull request merged**